scripts/generate_upgrade_syncset.py

```diff
@@ -0,0 +1,301 @@
+
+
+def get_cluster_name_for_id(cluster_id, environment):
+    """
+    Determines the clusterdeployment name associated with the supplied clusterid
+    :param cluster_id: cluster-id of cluster
+    :return: clusterdeployment name associated with cluster ID
+    """
+    cluster_ns = 'uhc-{}-{}'.format(environment, cluster_id)
+    try:
+        # pout = subprocess.check_output(['cat', '/home/mbargenq/scratch/cluster.txt']).decode(
+        #     'utf-8')
+        pout = subprocess.check_output(['oc', 'get', 'clusterdeployment', '-n', cluster_ns, '-o', 'json']).decode(
+            'utf-8')
+        output_as_dict = json.loads(pout)
+        if 'items' not in output_as_dict:
+            raise Exception('fetching clusterdeployment returned invalid response')
+
+        cd = output_as_dict['items']
+        if len(cd) == 0:
+            raise Exception('clusterdeployment not found')
+
+        return cd[0]['metadata']['name']
+
+    except Exception as err:
+        raise err
+
+
+def generate_upgradeconfig(start_time, version, channel):
+    """
+    Generates the UpgradeConfig CR in python dict form
+    :param version: version to upgrade to
+    :param channel: channel for upgrade
+    :param force: force upgrade
+    :return: dict representation of upgradeconfig CR
+    """
+    uc = {
+        'apiVersion': UC_API_VERSION,
+        'kind': 'UpgradeConfig',
+        'metadata': {
+            'name': UC_NAME,
+            'namespace': UC_NAMESPACE,
+        },
+        'spec': {
+            'type': dq('OSD'),
+            'upgradeAt': dq(start_time),
+            'proceed': True,
+            'PDBForceDrainTimeout': UC_PDB_DRAIN_TIMEOUT,
+            'desired': {
+                'version': dq(version),
+                'channel': dq(channel),
+                'force': UC_FORCE,
+            }
+        }
+    }
+    return uc
+
+
+def generate_syncset(upgradeconfig, cluster_id, cluster_name='', environment='production'):
+    """
+    Generates the SelectorSyncSet
+    :param upgradeconfig: UpgradeConfig CR as dict
+    :param cluster_id: Cluster ID to target
+    :param cluster_name: Cluster name to target
+    :return: SelectorSyncSet as dict
+    """
+
+    # generate the namespace for the SyncSet in hive format
+    # uhc-<environment>-<clusterID>
+    ss_namespace = 'uhc-{}-{}'.format(environment, cluster_id)
+
+    data = CM()
+    data['apiVersion'] = SS_API_VERSION
+    data['kind'] = 'SyncSet'
+    data['metadata'] = {
+        'labels': {
+            'api.openshift.com/id': cluster_id,
+            'api.openshift.com/name': cluster_name,
+        },
+        'name': SS_NAME,
+        'namespace': ss_namespace,
+    }
+
+    data['spec'] = {
+        'clusterDeploymentRefs': [
+            {
+                'name': dq(cluster_name),
+            }
+        ],
+        'resourceApplyMode': 'Upsert',
+        'resources': [
+            upgradeconfig
+        ],
+    }
+
+    yaml.preserve_quotes = True
```

Comment:
What is this line intending to do? (A comment please?)

Reply:
Linked to [`dq` above](https://github.com/openshift/managed-cluster-config/pull/409/files/f434cede5b9be15182902c779eb78de9860c58fd#diff-b2d12c252edac3e97ef460c9e9c1e4eaR12), it's intended to make the dumped YAML include any redundant quotes around strings, if they were present on input/creation. One reason you might want to do that ([example](https://gitlab.cee.redhat.com/service/app-interface/-/blob/1428b7c41f71d2811c4b6a20349b471f5e290116/hack/new_osd_operator/new_operator.py#L12)) is to minimize diffs when you're changing existing YAML documents that were (usually) written by hand. That doesn't apply here AFAICT.

Reply:
I've ported to PyYAML and removed the need for all of this in e1449f7